File: easy/1200_minimum_absolute_difference/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<std::vector<int>> minimumAbsDifference(std::vector<int> &arr) {
    std::sort(arr.begin(), arr.end());

    int minDifference = INT_MAX;
    std::vector<std::vector<int>> differences;

    for (size_t i = 1; i < arr.size(); ++i) {
      int currDifference = arr[i] - arr[i - 1];

      if (currDifference < minDifference) {
        differences.clear();
        minDifference = currDifference;
      }

      if (currDifference == minDifference)
        differences.push_back({arr[i - 1], arr[i]});
    }

    return differences;
  }
};

File: easy/1200_minimum_absolute_difference/solution.py
from typing import List


class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()

        minDifference = float("inf")
        differences = []

        for i in range(len(arr) - 1):
            currDifference = arr[i + 1] - arr[i]

            if currDifference == minDifference:
                differences.append((arr[i], arr[i + 1]))
            elif currDifference < minDifference:
                differences = [(arr[i], arr[i + 1])]
                minDifference = currDifference

        return differences
